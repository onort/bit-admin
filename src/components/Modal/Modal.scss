@import "../../styles/main.scss";

.container {
  position: fixed;
  z-index: 9999;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  background-color: rgba(0, 0, 0, 0.5);
}
