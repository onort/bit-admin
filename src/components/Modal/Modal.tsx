import React from "react"
import cx from "classnames"

import styles from "./Modal.scss"

interface Props {
  children: React.ReactNode
  className?: string
}

const Modal: React.SFC<Props> = props => {
  const className = cx(styles.container, props.className)
  return <div className={className}>{props.children}</div>
}

export default Modal
