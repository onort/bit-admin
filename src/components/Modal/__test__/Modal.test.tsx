import React from "react"
import { shallow } from "enzyme"

import Modal from "../"

const testMessage = "Test message."
const testClassName = "testClassName"

describe("<Modal />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Modal>
        <p>{testMessage}</p>
      </Modal>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(
      <Modal>
        <p>{testMessage}</p>
      </Modal>
    )
    expect(wrapper.find("p").text()).toBe(testMessage)
  })

  it("should be able to have a custom className", () => {
    const wrapper = shallow(
      <Modal className={testClassName}>
        <p>{testMessage}</p>
      </Modal>
    )
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
