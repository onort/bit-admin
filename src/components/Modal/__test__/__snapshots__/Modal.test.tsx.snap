// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should match snapshot 1`] = `
<div
  className="container"
>
  <p>
    Test message.
  </p>
</div>
`;

exports[`<Modal /> should match snapshot 2`] = `"<div class=\\"container\\"><p>Test message.</p></div>"`;
